wandb/sdk/launch/runner/kubernetes.py

```diff
@@ -303,7 +303,7 @@ def run(
         if not job_metadata.get("name"):
             job_metadata[
                 "generateName"
-            ] = f"launch-{launch_project.target_entity}-{launch_project.target_project}-"
+            ] = f"launch-{launch_project.target_entity.replace('_', '')}-{launch_project.target_project.replace('_', '')}-"
```

Comment:
Hmm, since this needs to be a valid DNS subdomain name, could we make this more general to follow these rules: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

Reply:
Also, lets replace the `_`s with `-`s to maintain some form of separator